### PR TITLE
PMM-7999 Add Quickstart guide

### PR DESCRIPTION
**Pull request by rasika-chivate**
Add quickstart guide for PMM

### CODE REVIEW

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# Quickstart Guide
+
+This guide helps you quickly get started with PMM using Docker.
```
Comment (alexfiru) [inline docs/quickstart.md:3]: Rephrase:
The procedures on this page are designed to guide you through the simplest way to get up and running with Percona Monitoring and Management.

File: docs/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+## System requirements
+
+| Disk      |Memory  | Ports         |
+|-----------|--------|---------------|
+| * 1 GB of storage per monitored database node. <br/> * 1 GB of storage per monitored database node for data retention set to one week.| Each database node should have at least 2 GB of memory for effective monitoring. <br/> The increase in memory usage is not proportional to the number of nodes. </br>  **Example**: Data from 20 nodes should be easily handled with 16 GB.| ***** By default, port 443 should be opened on the PMM Server. <br/> * The database port should be open for the PMM Agent.|
+
+## Install PMM
+
+To install the PMM server, follow the steps below:
```
Comment (alexfiru) [inline docs/quickstart.md:17]: Redundant.
"To install PMM Server:"

Comment (rasika-chivate) [reply docs/quickstart.md:14]: Fixed

File: docs/quickstart.md (hunk 1 above)
Comment (rasika-chivate) [reply docs/quickstart.md:3]: Fixed

File: docs/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    If you have access to the database node, install a PMM agent on it to reduce server-side resource utilization.
+
+
+=== "MySQL 5.7 or 8.0"
+
+    **Prerequisites**
+
+    Before you add a MySQL database for monitoring you should have a [database account for PMM](https://docs.percona.com/percona-monitoring-and-management/setting-up/client/mysql.html#create-a-database-account-for-pmm).
+
+    **Add a MySQL database instance**
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:72]: Fixed.

File: docs/quickstart.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+This guide helps you quickly get started with PMM using Docker.
+
+## Prerequisites
+
+Install [Docker](https://docs.docker.com/engine/install/).
+
+## System requirements
+
+| Disk      |Memory  | Ports         |
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:7]: List makes the page unnecessarily long.

File: docs/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
+# Quickstart Guide
+
+This guide helps you quickly get started with PMM using Docker.
+
+## Prerequisites
+
+Install [Docker](https://docs.docker.com/engine/install/).
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md]: I have removed the link as the install script installs docker as well.

File: docs/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+## Install PMM
+
+To install the PMM server, follow the steps below:
+{.power-number}
+
+1. Download and run the [PMM easy installation script](https://docs.percona.com/percona-monitoring-and-management/setting-up/server/easy-install.html) from [github](https://github.com/percona/pmm/blob/main/get-pmm.sh) to verify and install any missing software and dependencies. 
+
+    The install script only runs on Linux-compatible systems. To use it, run the command with sudo privileges or as root.
+    
+2. [Set up Docker on Mac](https://docs.docker.com/docker-for-mac/install) or [Linux](https://docs.docker.com/install/linux/docker-ce/ubuntu).
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md]: Removed this.

File: docs/quickstart.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+
+1. Download and run the PMM easy installation script from [github](https://github.com/percona/pmm/blob/main/get-pmm.sh). The install script only runs on Linux-compatible systems. To use it, run the command with `sudo` privileges or as root.
+
+    !!! note alert alert-primary ""
+        This script does the following:
+
+        - Installs Docker if it is not installed on your system.
+        - Stops and renames any currently running PMM Server Docker container from `pmm-server` to `pmm-server-{timestamp}`. This old pmm-server container is not a recoverable backup.
+        - Pulls and runs the latest PMM Server Docker image.
+        - Can run in `Interactive` mode to change the default settings:
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:24]: I have taken this from PMM QSG

File: docs/quickstart.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+
+## System requirements
+
+| Disk      |Memory  | Ports         |
+|-----------|--------|---------------|
+| * 1 GB of storage per monitored database node. <br/><br/>  * 1 GB of storage per monitored database node for data retention set to one week.| Each database node should have at least 2 GB of memory for effective monitoring. <br/><br/> **Note:** The increase in memory usage is not proportional to the number of nodes. </br> </br>  **Example**: Data from 20 nodes should be easily handled with 16 GB.| * By default, port 443 should be opened on the PMM Server. <br/><br/>  * The database port should be open for the PMM Agent.|
+
+## Install PMM
+
+The following steps will guide you through the installation of PMM server:
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:13]: Fixed

File: docs/quickstart.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        curl -fsSL https://www.percona.com/get/pmm | /bin/bash
+        ```
+
+    === "wget"
+
+        ```sh
+        wget -qO - https://www.percona.com/get/pmm | /bin/bash    
+        ```
+
+3. Log in to PMM with the default login credentials that are provided after the installation is completed.
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:47]: That adress is displayed after PMM is installed on your system.

File: docs/quickstart.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+        - Pulls and runs the latest PMM Server Docker image.
+        - Can run in `Interactive` mode to change the default settings:
+
+            ```sh
+            curl -fsSLO https://www.percona.com/get/pmm (or wget https://www.percona.com/get/pmm)
+            chmod +x pmm
+            ./pmm --interactive
+            ```
+    
+2. Install the PMM Server using `cURL` or `wget`:
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:32]: Fixed.

File: docs/quickstart.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+
+    === "wget"
+
+        ```sh
+        wget -qO - https://www.percona.com/get/pmm | /bin/bash    
+        ```
+
+3. Log in to PMM with the default login credentials that are provided after the installation is completed.
+
+    !!! note alert alert-primary "Note"
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:49]: Adding as a note would emphasize more.

[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/quickstart.md:49]: Fixed.

**Comment by rnovikovP** [comment]
- I'm not sure we need a step 1 at all. It's just an explanation of what the script is doing. Can this information be on the full page about the easy install script?  
- " database account for PMM" - can we embed the instructions on the page? 
- for  Mysql/PG/Mongo, we have only Remote monitoring options on the doc . This is not the most recommended way. Having node metrics - is very useful.   So let's also add the  selection "Do you have access to the DB host ?" option and based on  it have to Remove (as we have now) or instructions to  "Adding the client  and  adding DB using pmm-admin"


all this based on the assumption the goal of  this guide is: To be the Single page for the user to be able to add DB to PMM in most common cases with a majority of value added to the user

**Comment by rasika-chivate** [comment]
> * I'm not sure we need a step 1 at all. It's just an explanation of what the script is doing. Can this information be on the full page about the easy install script?
> * " database account for PMM" - can we embed the instructions on the page?
> * for  Mysql/PG/Mongo, we have only Remote monitoring options on the doc . This is not the most recommended way. Having node metrics - is very useful.   So let's also add the  selection "Do you have access to the DB host ?" option and based on  it have to Remove (as we have now) or instructions to  "Adding the client  and  adding DB using pmm-admin"
> 
> all this based on the assumption the goal of this guide is: To be the Single page for the user to be able to add DB to PMM in most common cases with a majority of value added to the user

@rnovikovP For point 3. Could you please provide the instructions as there's nothing wriiten in the docs.

File: docs/quickstart/index.md [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+        3. Register PMM Client:
+        
+            ```sh
+            pmm-admin config --server-insecure-tls --server-url=https://admin:admin@X.X.X.X:443
+            ```
+
+    6. Add the PostgreSQL database:
+
+        ```sh 
+        pmm-admin add postgresql --username=pmm --password=<your_password> POSTGRESQL_SERVICE
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:152]:
```suggestion
        pmm-admin add postgresql --username=pmm --password=<your_password>
```

File: docs/quickstart/index.md [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+            {
+                "role":"clusterMonitor",
+                "db":"admin"
+            },
+            {
+                "role":"read",
+                "db":"local"
+            }
+        ]
+        })
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:207]:
```suggestion
        db.getSiblingDB("admin").createUser({
            "user":"pmm",
            "pwd":"<your_password>",
            "roles":[
                {
                    "role":"explainRole",
                    "db":"admin"
                },
                {
                    "role":"clusterMonitor",
                    "db":"admin"
                },
                {
                    "role":"read",
                    "db":"local"
                }
            ]
        })
```

File: docs/quickstart/index.md [hunk 14]
```diff
@@ -0,0 +1,324 @@
...
+        exit
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:208]:
```suggestion
```

File: docs/quickstart/index.md [hunk 15]
```diff
@@ -0,0 +1,324 @@
...
+        3. [Register PMM Client]:
+        
+            ```sh
+            pmm-admin config --server-insecure-tls --server-url=https://admin:admin@X.X.X.X:443
+            ```
+            
+    3. Add the ProxySQL service:
+
+        ```
+        pmm-admin add proxysql --username=pmm --password=<your_password> PROXYSQL_SERVICE
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:273]:
```suggestion
        pmm-admin add proxysql --username=pmm --password=<your_password>
```

File: docs/quickstart/index.md [hunk 16]
```diff
@@ -0,0 +1,324 @@
...
+        3. Register PMM Client:
+                
+            ```sh
+            pmm-admin config --server-insecure-tls --server-url=https://admin:admin@X.X.X.X:443
+            ```
+
+    4. Run the command below, specifying the `listen-port`` as the port number where HAProxy is running. (This flag is mandatory.)
+
+        ```sh
+        pmm-admin add haproxy --listen-port=8404 HAPROXY_SERVICE
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:308]:
```suggestion
        pmm-admin add haproxy --listen-port=8404
```

File: docs/quickstart/index.md [hunk 17]
```diff
@@ -0,0 +1,324 @@
...
+        ```
+
+    For detailed instructions and more information on the command arguments, see the [HAProxy topic](../setting-up/client/haproxy.md).
+
+## Check database monitoring results
+
+After installing PMM And connecting the database, go to the database's Instance Summary dashboard. This shows essential information about your database performance and an overview of your environment.
```
Comment (artemgavrilov) [inline docs/quickstart/index.md:315]:
```suggestion
After installing PMM and connecting the database, go to the database's Instance Summary dashboard. This shows essential information about your database performance and an overview of your environment.
```

File: docs/quickstart/index.md [hunk 18]
```diff
@@ -0,0 +1,324 @@
...
+
+    To connect a PostgreSQL database: 
+    { .power-number}
+
+    1. Create a PMM-specific user for monitoring:
+        
+        ```
+        CREATE USER pmm WITH SUPERUSER ENCRYPTED PASSWORD '<your_password>';
+        ```
+
```
Comment (Nailya) [inline docs/quickstart/index.md:112]: also "pmm" user should be configured in pg_hba.conf file.
```
local   all             pmm                                md5
# TYPE  DATABASE        USER        ADDRESS                METHOD
```

File: docs/quickstart/index.md [hunk 19]
```diff
@@ -0,0 +1,324 @@
...
+            wget https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+            dpkg -i percona-release_latest.generic_all.deb
+            ```
+        2. Install the PMM Client package:
+            
+            ```sh
+            apt update
+            apt install -y pmm2-client
+            ```
+
```
Comment (Nailya) [inline docs/quickstart/index.md:83]: I think we should mention that there is another way of installation for red-hat based OSes.

File: docs/quickstart/index.md [hunk 20]
```diff
@@ -0,0 +1,324 @@
...
+            ```sh
+            wget https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+            dpkg -i percona-release_latest.generic_all.deb
+            ```
+        2. Install the PMM Client package:
+            
+            ```sh
+            apt update
+            apt install -y pmm2-client
+            ```
```
Comment (Nailya) [inline docs/quickstart/index.md:142]: these command work only for debian-based OSes and we should mention red-hat based command also.

**Comment by mmnosek** [comment]
I love the improvements! A few comments after having a quick look:
1. I would change "Install PMM" headline to "Install PMM Server"
2. I would change "Once PMM is set up, choose the database that you want it to monitor:" to "Once PMM Server is set up, choose the service that you want it to monitor" (technically, we also monitor ProxySQL and HAProxy which are not DBs
3. Installing PMM package step in each tech would benefit from having two tabs - with Debian and RHEL commands (apt-get vs yum).
4. "Install Percona Release Tool" should be linked ti its documentation
5. "Alternative database connection workflows" tab has "PMM Client installation via User Interface" link. I think it would be better to rename it to "Adding remote instance via User Interface" (because technically it does not install PMM Client anywhere)
6. Postgres instructions are incomplete. They miss steps required to enable pg_stat_monitor. 
7. Step 2 of POstgres instructions are too long. I don't think all the explanation what pg_stat_monitor is is required.

**Comment by rnovikovP** [comment]
> 1. I would change "Install PMM" headline to "Install PMM Server"
This was intentional (my ask) to keep users from overloading about Client-server and options. 
As the main action they expect - is "install PMM," not "some pmm components,"

> 2. I would change "Once PMM is set up, choose the database that you want it to monitor:" to "Once PMM Server is set up, choose the service that you want it to monitor" (technically, we also monitor ProxySQL and HAProxy which are not DBs

Perhaps we should label it as "Database or Application" instead. Terms like Services and Nodes are internal jargon, and using PMM is the best way to get familiar with them. Our goal is to use language that is easily understandable to users in our initial interactions.


Will all other suggestions - 👍

**Pull request merged**